### ISSUE TITLE
FiwixOS 2 Live Floppy (1.44MB) with initrd using RAMdisk driver - is booting fine :)

### ISSUE THREAD

**Issue by informer2016** [mikaku/Fiwix#3]
Hi there @mikaku , just wanted to tell you that I've tested a _"FiwixOS 2 Live Floppy (1.44MB) with initrd using RAMdisk driver"_ floppy and it's still booting fine as a part of coreboot+SeaBIOS. Wish you the happy and productive times ahead ;-)

**Comment by mikaku** [comment]
Good to know! :smiley: 

Thank you very much.

**Comment by informer2016** [comment]
Good day @mikaku , nice to inform you that a newer 16-Dec-2020 Fiwix 3 Live RAMdisk floppy is also working fine as a part of coreboot+SeaBIOS! Although some older bugs are still there _(i.e. no hard drives visible and a reboot doesn't work)_, I get it may be hard to fix them if you don't have a similar hardware - and really glad to see your recent important commits & bugfixes. Wish you all the best on your path - and, if you would like me to test something specific in more detail, just tell and I will happily assist you ;-)

**Comment by mikaku** [comment]
> Good day @mikaku , nice to inform you that a newer 16-Dec-2020 Fiwix 3 Live RAMdisk floppy is also working fine as a part of coreboot+SeaBIOS! 

That's great news! :smiley: 

>Although some older bugs are still there (i.e. no hard drives visible and a reboot doesn't work), I get it may be hard to fix them if you don't have a similar hardware

Are those hard drives under a SATA controller? remember that Fiwix only supports IDE/ATA.

Strange that reboot doesn't work. I've been tested it successfully on several machines (real hardware) and also on QEMU, VMware and Bochs. Is the procedure to reboot different on machines based on coreboot+SeaBIOS?

> if you would like me to test something specific in more detail, just tell and I will happily assist you ;-)

It is possible that I can test your coreboot+SeaBIOS environment here?

> and really glad to see your recent important commits & bugfixes.

Yeah, I'm having a very productive year with Fiwix.

Now I'm working on the new framebuffer console using VESA VBE 2.0. Since I use GRUB 0.97 I needed to patch it to support the GRUB commands `vbeset` and `vbematch` to be able switch to the desired graphics mode. I discarded to use GRUB 2 because I'm not sure that it will fit on a floppy disk, and if so, I don't think there will be enough space left for Fiwix.
 
It's really nice to have a floppy (1.44MB) capable of booting with a framebuffer console (1024x768x32) with 128 columns and 48 rows. It's a more comfortable experience. :wink:

**Comment by informer2016** [comment]
> Are those hard drives under a SATA controller? remember that Fiwix only supports IDE/ATA.

Yes they are SATA, unfortunately _(for Fiwix support)_

> Strange that reboot doesn't work. I've been tested it successfully on several machines (real hardware) and also on QEMU, VMware and Bochs. Is the procedure to reboot different on machines based on coreboot+SeaBIOS?

Seeing that you got a reboot working with QEMU, which has its' own built-in coreboot+SeaBIOS actually - this seems to be a hardware-specific bug, maybe not coreboot-related but AMD related. So, although I could provide a special coreboot+SeaBIOS build for QEMU with a Fiwix floppy inside, it's unlikely it would help - and would be much more useful if you have an access to some AMD-based PC and could debug Fiwix there.

> Since I use GRUB 0.97 I needed to patch it to support the GRUB commands vbeset and vbematch to be able switch to the desired graphics mode.

Really nice. Did you consider sending this patch to GRUB developers, to be included in a potential "0.98" release?

**Comment by mikaku** [comment]
> ... and would be much more useful if you have an access to some AMD-based PC and could debug Fiwix there.

Yes I did it, specifically on an old AMD K6 (*3D processor*) at 300Mhz. FiwixOS 3 was installed there successfully and it reboots the PC with no problems at all.

> Really nice. Did you consider sending this patch to GRUB developers, to be included in a potential "0.98" release?

That patch is not mine, I just downloaded it directly from [here](https://forum.osdev.org/download/file.php?id=915&sid=f323f373a98acad563ae718daf863712), from this [OSDev Forum thread](https://forum.osdev.org/viewtopic.php?t=18255).

According the ChangeLog in the patch, the author was Durand J. Miller (and Brynet-Inc tried to include it in the 0.97 version?):

```
+2006-11-23  Brynet-Inc <brynet@gmail.com>
+
+	* updated grub 0.97 with a VBE patch by Durand J. Miller 
+	<djmiller@djm.co.za> from 2004-05-09.
[...]
```

Also, there is [this thread in GNU Mailing List](https://lists.gnu.org/archive/html/bug-grub/2004-01/msg00005.html) where you can see that Durand Miller attempted (unsuccessfully) to include his patch in GRUB 0.97.

Here is a preview floppy image of the Fiwix kernel that includes support of VESA VGA frame buffer on a patched GRUB 0.97:

[fiwix-floppy.img.gz](https://github.com/mikaku/Fiwix/files/6305217/fiwix-floppy.img.gz)
